internal/core/src/common/EasyAssert.h

```diff
@@ -18,66 +18,97 @@
 #include <string_view>
 #include <stdexcept>
 #include <exception>
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 #include <string>
 #include "pb/common.pb.h"
 #include "common/type_c.h"
 
 /* Paste this on the file if you want to debug. */
 namespace milvus {
-enum ErrorCodeEnum {
+enum ErrorCode {
     Success = 0,
     UnexpectedError = 1,
-    BuildIndexError = 2,
-    IllegalArgument = 5,
+    NotImplemented = 2,
+    IndexBuildError = 3,
+    IndexAlreadyBuild = 4,
+    DataTypeInvalid = 6,
+    LocalChunkManagerError = 7,
+    InvalidPathError = 8,
+    OpenFileError = 9,
+    CreateFileError = 10,
+    ReadFileError = 11,
+    WriteFileError = 12,
+    PathAlreadyExistError = 13,
+    DirNotExistError = 14,
+    BucketInvalid = 15,
+    ObjectNotExist = 16,
+    S3Error = 17,
+    RetrieveError = 18,
+    FieldIDInvalid = 19,
+    FieldAlreadyExist = 20,
+    OpTypeInvalid = 21,
+    DataIsEmpty = 22,
+    JsonKeyInvalid = 23,
+    UnistdError = 30,
+    KnowhereError = 100,
```

Comment:
we need new codes for segcore (like start from 2000), then milvus can recognize them and handle them